src/Compiler/RefC/RefC.idr

```diff
@@ -1044,27 +1031,18 @@ executeExpr c _ tm
          coreLift_ $ system "false"
 
 export
-compileExpr : UsePhase
-           -> Ref Ctxt Defs
-           -> (tmpDir : String)
-           -> (outputDir : String)
-           -> ClosedTerm
-           -> (outfile : String)
-           -> Core (Maybe String)
-compileExpr ANF c _ outputDir tm outfile =
-  do let outn = outputDir </> outfile ++ ".c"
-     let outobj = outputDir </> outfile ++ ".o"
-     let outexec = outputDir </> outfile
-     coreLift_ $ mkdirAll outputDir
-     cdata <- getCompileData False ANF tm
-     let defs = anf cdata
-     _ <- newRef ArgCounter 0
+generateCSourceFile : {auto c : Ref Ctxt Defs}
+                   -> List (Name, ANFDef)
+                   -> (outn : String)
+                   -> Core (Maybe String)
+generateCSourceFile defs outn =
+  do _ <- newRef ArgCounter 0
      _ <- newRef FunctionDefinitions []
      _ <- newRef TemporaryVariableTracker []
      _ <- newRef OutfileText DList.Nil
      _ <- newRef ExternalLibs []
      _ <- newRef IndentLevel 0
-     traverse_ (\(n, d) => createCFunctions n d) defs
+     _ <- traverse (\(n, d) => createCFunctions n d) defs
```

Comment:
What's the reason behind this change?

Reply:
This change comes from merging `master`. You'd have to ask whoever made the original commit.

Reply:
`Core.Core.traverse` in this case is `Core (List Unit)`, and it seems that to sequence something it needs to return `Core Unit`.

Reply:
Right so I suspect that the conflict was resolved the wrong way.

Reply:
I see. Let me give it a go on a fresh branch.

Reply:
No, manually inserting those changes on a fresh branch gives the same error. Let's chat.